### ISSUE TITLE
UnnecessaryAnnotationUseSiteTarget rule

### ISSUE THREAD

**Issue by vanniktech** [arturbosch/detekt#1138]
Quite often I see that the annotation use site targets are used inappropriately. 

For instance:

```kotlin
class Presenter(
  ...
  @param:SchedulerMain private val mainScheduler: Scheduler,
  @param:SchedulerComputation private val computationScheduler: Scheduler
)
```

`param:` isn't required and hence it should be flagged.

**Comment by schalkms** [comment]
Looks like an idea for a new rule :)
Could you please explain this issue a bit more detailed or give a reference?
Is this related to a specific framework?

**Comment by vanniktech** [comment]
It's just an annotation applied to a parameter of a constructor. The use site isn't required though.

**Comment by arturbosch** [comment]
Oha, never used this feature :O

**Comment by vanniktech** [comment]
If you're using Dagger and you're injecting the same type and need a qualifier you need it. e.g. `@Inject @field:MyQualifier lateinit var foo: Foo`. This confuses people though so they feel like they also need that for normal parameter declarations.

**Comment by BraisGabin** [comment]
As far as I can see in the documentation https://kotlinlang.org/docs/reference/annotations.html#annotation-use-site-targets if you don't set any target `param` is the first default one so we could remove it always safely.

Something similar happens with `property`. If the property is not declared in as a parameter of the constructor we could remove it too.

I'll give it a try.
